### PR TITLE
Assertions!

### PR DESCRIPTION
**Pull request by inikulin**
I'll add motivation and feature summary later.

\cc @AlexanderMoskovkin @AndreyBelym @helen-dikareva @georgiy-abbasov

### CODE REVIEW

**Comment by inikulin** [comment]
Actually everyone is welcome to review. \cc @DevExpress/testcafe

**Comment by inikulin** [comment]
@testcafe-build-bot \retest

File: test/functional/fixtures/api/es-next/assertions/testcafe-fixtures/assertions-test.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    await t
+        .expect(32).lte(42)
+        .expect(42).lte(42)
+        .expect(42).lte(12);
+});
+
+test('.within() assertion', async t => {
+    await t
+        .expect(2.3).within(2, 3)
+        .expect(4.5).within(4.6, 7);
```
Comment (helen-dikareva) [inline test/functional/fixtures/api/es-next/assertions/testcafe-fixtures/assertions-test.js]: Maybe we should add check of border conditions to be sure that the boundaries are included

**Comment by kirovboris** [comment]
I like this features, but ```gte``` (greater or equal?) and ```lte``` abbreviations look a little bit strange in my opinion.

File: test/functional/fixtures/api/es-next/assertions/testcafe-fixtures/assertions-test.js (hunk 1 above)
Comment (inikulin) [inline test/functional/fixtures/api/es-next/assertions/testcafe-fixtures/assertions-test.js]: `include` is not the same as `within`. It's an alias for `contains`: it checks that actual value contains some value. `within` checks that actual value is within some range.

**Comment by inikulin** [comment]
@kirovboris 

> I like this features, but gte (greater or equal?) and lte abbreviations look a little bit strange in my opinion.

It's pretty common, e.g.: https://docs.mongodb.com/manual/reference/operator/query/lte/. The alternatives are: `gtOrEql` and `ltOrEql` or `atLeast` and `atMost`. TBH, I don't like both.

**Comment by inikulin** [comment]
@helen-dikareva 

> /r-
great work)

😆

File: test/functional/fixtures/api/es-next/assertions/testcafe-fixtures/assertions-test.js (hunk 1 above)
Comment (helen-dikareva) [inline test/functional/fixtures/api/es-next/assertions/testcafe-fixtures/assertions-test.js]: oh, I see

File: src/cli/index.js [hunk 2]
```diff
@@ -102,6 +102,7 @@ function useLocalInstallation () {
     var local = resolveCwd('testcafe/lib/cli');
 
     if (local && local !== __filename) {
+        log.write('Using locally installed version of TestCafe.');
```
Comment (AlexanderMoskovkin) [inline src/cli/index.js]: @inikulin  Just to be sure, is this line should be there? (I have this question because the line is not related to assertions)

Comment (inikulin) [inline src/cli/index.js]: Yes, just decided to not create additional PR for this one.

File: test/functional/fixtures/api/es-next/assertions/testcafe-fixtures/assertions-test.js (hunk 1 above)
Comment (inikulin) [inline test/functional/fixtures/api/es-next/assertions/testcafe-fixtures/assertions-test.js]: But thinking about it a little bit more, I guess we shouldn't because:
1. There is no need to test chai functionality, it's already well tested.
2. It will require to modify callsite for other tests and I'm too lazy to do that 😋

**Comment by inikulin** [comment]
FPR

**Comment by georgiy-abbasov** [comment]
afaik we can use SelectorResultPromise only as 'actual' argument, not as 'expected', in nutshell we can't write something like next:
test('.eql() assertion', async t => {
    const el1 = Selector('#el1');
    const el2 = Selector('#el2');

    await t.expect(el1.getStyleProperty('float')).eql(el2.getStyleProperty('float'));
});
Just want to know, what do you think about it ?

**Comment by inikulin** [comment]
@georgiy-abbasov Is there any real life scenarios for this? The common case is to compare with some constant value or with previous state of element.

**Comment by georgiy-abbasov** [comment]
@inikulin, irl i saw when mixed up 'actual' and 'expected' (not here), only it.

**Comment by inikulin** [comment]
@georgiy-abbasov I guess we can go with this implementation for now and once where will be a demand implement retries for the actual value as well.

**Comment by kirovboris** [comment]
Are we calculating ```Selector``` automatically in the following case:
```javascript
const firstElm     = Selector('div:first-child');
const getActiveElm = Selector('div.active');

expect(firstElm).eql(getActiveElm);
```

**Comment by helen-dikareva** [comment]
No, now to execute them you should call
```
expect(await firstElm()).eql(await getActiveElm());
```
